web/src/store/dandiset.js

```diff
@@ -66,8 +65,10 @@ export default {
     async fetchPublishDandiset({ state, commit }, { identifier, version }) {
       state.loading = true;
 
+      const sanitizedVersion = version || (await publishRest.mostRecentVersion(identifier)).version;
```

Comment:
It could be, but I don't see how it would cause issues. `draft` is a valid version string in the django API (published versions look like `0.1234.123456` or something).